--- origsrc/vala-0.3.5/tests/testrunner.sh	2008-08-15 14:10:43.790570300 -0500
+++ src/vala-0.3.5/tests/testrunner.sh	2008-08-15 14:08:15.253202300 -0500
@@ -33,7 +33,7 @@
 export G_DEBUG=fatal_warnings
 
 VALAC=$topbuilddir/compiler/valac
-CC="gcc -std=c99"
+CC="gcc -std=gnu99"
 CFLAGS="-O0 -g3 -I$topsrcdir -I$topbuilddir"
 LDLIBS="-lm ../gee/.libs/libgee.a"
 
@@ -50,7 +50,7 @@
 		CODE=1
 		continue
 	fi
-	if ! $CC $CFLAGS $testbuild.c $(pkg-config --cflags --libs gobject-2.0) -o $testbuild $LDLIBS > $testbuild.err 2>&1
+	if ! $CC $CFLAGS $testbuild.c $LDLIBS $(pkg-config --cflags --libs gobject-2.0) -o $testbuild > $testbuild.err 2>&1
 	then
 		echo "ERROR: Compiling" $testbuild.c
 		cat $testbuild.err
@@ -62,7 +62,7 @@
 		rm $testbuild.c $testbuild.h $testbuild$exe $testbuild.err
 	else
 		echo "ERROR: test failed. This is the difference between" $testbuild.exp "and" $testbuild.err
-		diff -u $testbuild.exp $testbuild.err
+		diff -u $testsrc.exp $testbuild.err
 		CODE=1
 	fi
 done
